### ISSUE TITLE
Hive

### ISSUE THREAD

**Issue by JeffreyStrahm** [siggame/MegaMinerAI-Game-Ideas#29]
EDIT: I decide that simultaneous is probably a bit too much, at least with the step system, the one person acting "First" wouldn't be as much of an advantage if that system didn't exist.

Edit 2: Added a picture example. Removed food spawning at the beginning, so now there is a phase 1 build base, phase 2 kill enemy queen, however, it won't be as bad as before where you would want to try to gather food. The builders would mainly be focusing on digging out where you want your base to be and setting everything up, which they will have to continue to do, due to every unit in the game being able to dig through it.

Edit 3: Players no longer start with 2 worker ants but with just a queen with no food.

Edit 4: Reduced the abilities to more accuratly show the number of extra flags and got rid of large lungs. Also added some forggotton details like you can't walk over food.

First off, I just want to say I am bringing this over from the last megaminer, but I have hugely overhauled the game. It should be simpler now (relatively speaking). Give me feedback and tell me what you think!

**# Story:**
It is just after the nuptial flight season and you are a new queen ant looking to burrow into the ground to start your colony. You have found truly prime real-estate, however, you aren't alone. A new queen from another colony is setting up next door! This will surely lead to war. Which queen will survive and get to keep this land for themselves?

# **TLDR**
- turn based. see **movement**
- Kill the enemy queen to win. Each person get's 1, and only 1, queen. see **victory**
- Start with few units build up. see **victory**
- 5 unit types. See **units**
     1. fast, weak, flimsy, cheap
     2. slow, weak, tanky, mid-price
     3. mid-speed, strong, mid-defence, mid-price
     4. slow, strong, flimsy, pricy, ranged
     5. mid-speed, weakest, flimsy, cheap, builder
- procedurally generated terrain (moister level, see **terrain**)
- unit exponential cost. see **food**
- resources for only producing units. See **units** and **food**

# **Victory**
To win you must simply kill the enemy queen.
If victory has not been reached in 400 cycles (see **movement**) the victor will be determined by who has dealt the most damage to the enemy queen.
If they are tied for that, victory will be decided by who has spent the most on units.
If they are tied for that, victory will be decided by who has killed the highest value in units.
If they are still tied, victory will be decided by who has gathered the most food.
If they are still somehow tied, provided they actually did anything, victory will be decided by a coin flip.

### start:
Each team starts with their queen and no food. This will make them focus on digging in early on while they wait for their natural food production to give them food.

# **Food**
Food is a resource that naturally spawns in the terrain in a mirrored, random fashion. It is also produced by Queen to stop snowballing from raids where the opponent's food is all taken or from poor early gathering. Each food grows up one at a rate of (10/moisture level - 1) cycles, rounded up, as long as the tile is not over 5 moister. Food is used to buy units.

# **Units**
A units cost is determined by the base cost out into this equation for their own pool. The pools are builders, and melee and ranged. (base+[{units/2}^2/{queen's moisture}^.5]). Caps at 6. Minimum cost is the base cost. The cost is subtracted from the queen's stockpile.

## Stats:
HP: hit points. 0 or below and they die.
Damage: Amount they subtract from HP of their opponent.
Def: Damage reduction. (max 2)
speed: steps they act in a cycle (max: 4)
cost: price to build (min 1)
abilities: special modifiers

### **queen**
HP: 100
Damage: 4
Def: 0
Speed: 2
Dig-speed: .5 dirt per step
item capacity: unlimited food.
attacks per cycle: 2.
Cost: N/A
Abilities: Unit creation, Food recycling.

### **scout** (melee group)
HP: 10
Damage: 3
Def: 0
Speed: 4
Dig-speed: .125 dirt per step
item capacity: 0
attacks per cycle: 1.
Cost: 1
Abilities: N/A

### **tank** (melee group)
HP: 10
Damage: 3
Def: 2
Speed: 1
Dig-speed: 1 dirt per step
item capacity: 1
attacks per cycle: 1.
Cost: 2
Abilities: N/A

### **warrior** (melee group)
HP: 10
Damage: 5
Def: 1
Speed: 2
Dig-speed: .5 dirt per step
item capacity: 0.
attacks per cycle: 1.
Cost: 2
Abilities: N/A

### **spitter** (ranged group)
HP: 10
Damage: 7
Def: 0
Speed: 1
Dig-speed: .5 dirt per step
item capacity: 0.
attacks per cycle: 1.
Cost: 3
Abilities: ranged

### **worker** (builder group)
HP: 5
Damage: 0
Def: 0
Speed: 2
Dig-speed: 1 dirt per step
item capacity: 3 of a single type.
attacks per cycle: 0.
Cost: 1
Abilities: builder

## Abilities: These are simply additional modifiers to what they can do.
**unit creation:** Can create units. (when units are created, spawn them around this unit)
**Food recycling:** gets .25 food per cycle. (add .25 food to what this and is carrying.) (rubberbanding)
**ranged:** May attack over a distance (3 range)
**builder:** May plant food, and build walls.

# **movement**
Every unit has a "speed". This is the amount of steps they act in, in a cycle. All units, unless stated otherwise, can only attack once a cycle, but they may move every step.
speed 4: + + + +
speed 3: - + + +
speed 2: - + - +
speed 1: - - - +
Attacks may take place at the start or end of a step that the unit can act on.
Note, unit death is calculated **AFTER** both teams have moved that step. That way the first turn advantage isn't too much.

# **Terrain**
The map will be 41 by 26. Can add a wall around the map to guard against the greatest enemy. Seg faults. Like the rail in saloon.
The farthest 7 tiles at the top and bottom will be dirt. Every tile gets a randomly generated moisture value, listed from most common to rarest: 2, 1, 3, 4, 5, 6, 10
10 moisture tiles are water, and if a unit stays in the tile at the end of the cycle, they one point of damage. Defense doesn't reduce this. If a unit is in a tile of 3 moisture or more, at the end of the cycle, they regain ([mosture-2]*.5) hit points. 

**stockpiles:** (can be walked onto) (not a actual tile type)
there may be a maximum of 10 of one type of resource in every tile. These do not effect movement. Not built, just true for every tile

**Items:**
When a food is harvest/walls dug, the item will try to drop at the units feet. A unit must be on the same tile as the item to pick it up. and it costs a move to do so.

**Walls/dirt:** (can't be walked onto)
A naturally spawning wall starts off with 3 dirt. A unit must dedicate one cycle to digging at a wall, base, to remove one dirt. (scouts take 2 cycles, builders remove 1 dirt per step). A unit with the builder ability may use 1 dirt from their personal hold to make a tile dirt, or add a dirt to a wall. There is no cap.

**food:** food as a value of 1 when planted, and will grow by one with a maximum of 3. Any unit may "dig" up one food per step. You can't walk on them.

A code explanation of the digging. The unit must reach 1 to remove a dirt. A scout subtracts .125, a builder subtract 1 for example. This is per "step".

![new_ant](https://cloud.githubusercontent.com/assets/22081179/22724558/a9be793e-ed8c-11e6-985a-a2d842f25969.png)
